installinstallmacos.py

```diff
@@ -30,42 +30,110 @@
+    ioreg_cmd = ['ioreg', '-p', 'IODeviceTree', '-r', '-n', '/', '-d', '1']
+    try:
+        ioreg_output = subprocess.check_output(ioreg_cmd).splitlines()
+        for line in ioreg_output:
+            if 'board-id' in line:
+                board_id = line.split(" ")[-1]
+                board_id = board_id[board_id.find('<"')+2:board_id.find('">')]
+                return board_id
+    except subprocess.CalledProcessError, err:
+        raise ReplicationError(err)
+
+
+def is_a_vm():
+    '''Determines if the script is being run in a virtual machine'''
+    sysctl_cmd = ['/usr/sbin/sysctl', 'machdep.cpu.features']
+    try:
+        sysctl_output = subprocess.check_output(sysctl_cmd)
+        cpu_features = sysctl_output.split(" ")
+        is_vm = False
+        for i in range(len(cpu_features)):
+            if cpu_features[i] == "VMM":
+                is_vm = True
+    except subprocess.CalledProcessError, err:
+        raise ReplicationError(err)
+    return is_vm
+
+
+def get_hw_model():
+    '''Gets the local system ModelIdentifier'''
+    sysctl_cmd = ['/usr/sbin/sysctl', 'hw.model']
+    try:
+        sysctl_output = subprocess.check_output(sysctl_cmd)
+        hw_model = sysctl_output.split(" ")[-1].split("\n")[0]
+    except subprocess.CalledProcessError, err:
+        raise ReplicationError(err)
+    return hw_model
+
+
+def get_current_build_info():
+    '''Gets the local system build'''
+    build_info = []
+    sw_vers_cmd = ['/usr/bin/sw_vers']
+    try:
+        sw_vers_output = subprocess.check_output(sw_vers_cmd).splitlines()
+        for line in sw_vers_output:
+            if 'ProductVersion' in line:
+                build_info.insert(0, line.split("\t")[-1])
+            if 'BuildVersion' in line:
+                build_info.insert(1, line.split("\t")[-1])
+    except subprocess.CalledProcessError, err:
+        raise ReplicationError(err)
+    return build_info
+
+
 def get_seeding_program(sucatalog_url):
     '''Returns a seeding program name based on the sucatalog_url'''
     try:
         seed_catalogs = plistlib.readPlist(SEED_CATALOGS_PLIST)
         for key, value in seed_catalogs.items():
             if sucatalog_url == value:
                 return key
+        return ''
     except (OSError, ExpatError, AttributeError, KeyError):
-        return None
+        return ''
 
 
-def get_seed_catalog():
+def get_seed_catalog(seedname='DeveloperSeed'):
     '''Returns the developer seed sucatalog'''
     try:
         seed_catalogs = plistlib.readPlist(SEED_CATALOGS_PLIST)
-        return seed_catalogs.get('DeveloperSeed', DEFAULT_SUCATALOG)
+        return seed_catalogs.get(seedname)
     except (OSError, ExpatError, AttributeError, KeyError):
-        return DEFAULT_SUCATALOG
+        return ''
+
+
+def get_default_catalog():
+    '''Returns the default softwareupdate catalog for the current OS'''
+    darwin_major = os.uname()[2].split('.')[0]
+    return DEFAULT_SUCATALOGS.get(darwin_major)
```

Comment:
again this is all in master -- not clear why this highlighted as a change.

Reply:
As above.

Reply:
OK, but complicates reviewing your changes since I have to double-check against master to see what's "really" changed.